### PR TITLE
feat!: Added support for propagate_tags feature

### PR DESCRIPTION
**Pull request by huzaifa-binafzal**
## Description
Updated the module to support propagate_tags variable inside replica_regions variable that we already have.

## Motivation and Context
Currently we're facing the issue where there is no way to propagate tags to the DynamoDB replicas table in other region.
This PR resolve the issue reported here: https://github.com/terraform-aws-modules/terraform-aws-dynamodb-table/issues/58

## Breaking Changes
This change does not break anything.

## How Has This Been Tested?
- [x] I have updated at least one of the `examples/*` to demonstrate and validate my change(s)
- [x] I have tested and validated these changes using one or more of the provided `examples/*` projects
- [x] A replica table was created and tags were propagated when propagate_tags value was set to true.  All went well.

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -58,8 +58,9 @@ resource "aws_dynamodb_table" "this" {
     for_each = var.replica_regions
 
     content {
-      region_name = replica.value.region_name
-      kms_key_arn = lookup(replica.value, "kms_key_arn", null)
+      region_name    = replica.value.region_name
+      kms_key_arn    = lookup(replica.value, "kms_key_arn", null)
+      propagate_tags = lookup(replica.value, "propagate_tags", false)
```
Comment (antonbabenko) [inline main.tf:63]: Usually, setting null as the default is preferred to avoid changes for existing users. Please change it.

**Review by antonbabenko — CHANGES_REQUESTED** [review]
Please update minimum version of AWS provider in versions.tf and in example to be from 4.23 since this argument was added then. So, this is going to be a major release.

File: main.tf (hunk 1 above)
Comment (huzaifa-binafzal) [reply main.tf]: Please review now.

**Review by antonbabenko — COMMENTED** [review]
LGTM. Run "pre-commit run -a" and commit the changes to make CI checks green.

**Comment by huzaifa-binafzal** [comment]
> LGTM. Run "pre-commit run -a" and commit the changes to make CI checks green.

@antonbabenko Please review now

**Comment by antonbabenko** [comment]
This PR is included in version 3.0.0 :tada:

**Pull request merged**